Sources/ComposableArchitectureMacros/ViewActionMacro.swift

```diff
@@ -0,0 +1,158 @@
+@preconcurrency import SwiftDiagnostics
+import SwiftSyntax
+import SwiftSyntaxMacroExpansion
+import SwiftSyntaxMacros
+
+public struct ViewActionMacro: MemberMacro {
+  public static func expansion<Declaration, Context>(
+    of node: SwiftSyntax.AttributeSyntax,
+    providingMembersOf declaration: Declaration,
+    in context: Context
+  ) throws -> [SwiftSyntax.DeclSyntax]
+  where Declaration: SwiftSyntax.DeclGroupSyntax, Context: SwiftSyntaxMacros.MacroExpansionContext {
+    guard declaration.hasStoreVariable
+    else {
+      // TODO: Fix it to add `let store: StoreOf<<#Feature#>>`?
+      context.diagnose(
+        Diagnostic(
+          node: declaration,
+          message: MacroExpansionErrorMessage(
+            """
+            @ViewAction macro requires \
+            \(declaration.identifierDescription.map { "'\($0)' " } ?? "") to have a 'store' \
+            property of type 'Store'.
+            """
+          )
+        )
+      )
+      return []
+    }
+
+    declaration.diagnoseDirectStoreDotSend(
+      declaration: declaration,
+      context: context
+    )
+
+    guard
+      case let .argumentList(arguments) = node.arguments,
+      arguments.count == 1
+    else { return [] }
+    guard
+      let memberAccessExpr = arguments.first?.expression.as(MemberAccessExprSyntax.self)
+    else { return [] }
+    let rawType = String("\(memberAccessExpr)".dropLast(5))
+
+    return [
+      """
+      fileprivate func send(_ action: \(raw: rawType).Action.View) {
+        self.store.send(.view(action))
+      }
+      fileprivate func send(_ action: \(raw: rawType).Action.View, animation: Animation?) {
+        self.store.send(.view(action), animation: animation)
+      }
+      """
+    ]
+  }
+}
+
+struct SimpleDiagnosticMessage: DiagnosticMessage, Error {
+  let message: String
+  let diagnosticID: MessageID
+  let severity: DiagnosticSeverity
+}
+
+extension SimpleDiagnosticMessage: FixItMessage {
+  var fixItID: MessageID { diagnosticID }
+}
+
+extension SyntaxProtocol {
+  func diagnoseDirectStoreDotSend<D: SyntaxProtocol>(
+    declaration: D,
+    context: some MacroExpansionContext
+  ) {
+    for decl in declaration.children(viewMode: .fixedUp) {
+      if let memberAccess = decl.as(MemberAccessExprSyntax.self),
+        let identifierSyntax = memberAccess.base?.as(DeclReferenceExprSyntax.self),
+        identifierSyntax.baseName.text == "store",
+        memberAccess.declName.baseName.text == "send"
+      {
+        context.diagnose(
+          Diagnostic(
+            node: decl,
+            message: MacroExpansionWarningMessage(
+              """
+              Do not use 'store.send' directly when using @ViewAction. Instead, use 'send'.
+              """
+            ),
```

Comment:
I went with a warning for these just in case we mis-diagnose something so it does not break people's builds. We could strengthen in the future if we are more comfortable with the diagnostics.